src/components/EditTextModal/EditTextModal.js

```diff
@@ -0,0 +1,156 @@
+    //   ]
+    // }
+    // convert to a flat structure
+    // textCoordinates = [{x1:32, x2:1, y1:582, y2:2, pageNum:1}, {x1:1, x2:1, y1:2, y2:2, pageNum:1}]
+
+    if (textCoordinates) {
+      const convertedTextCoordinates = [];
+      Object.keys(textCoordinates).forEach(pageNum => {
+        textCoordinates[pageNum].forEach(rect => {
+          const point1 = window.docViewer.getDocument().getPDFCoordinates(pageNum, rect.x1, rect.y1);
+          const point2 = window.docViewer.getDocument().getPDFCoordinates(pageNum, rect.x3, rect.y3);
+          convertedTextCoordinates.push({ x1: point1.x, x2: point2.x, y1: point1.y, y2: point2.y,pageNum:Number(pageNum) });
+        });
+      });
+      setTextCoordinates(convertedTextCoordinates);
+    }
+  };
+
+  const updateText = e => {
+    e.preventDefault();
+    window.PDFNet.runWithCleanup(async() => {
+      const doc = await window.docViewer.getDocument().getPDFDoc();
+      // Treat each line as a rectangle region and update them separately
+      const lines = newText.split(/\r?\n/);
+      doc.initSecurityHandler();
+      doc.lock();
+      const PDFNet = window.PDFNet;
+      // Note: If the number of lines are more than number of rects defined in textCoordinates, then we just ignore the rest
+      if (textCoordinates && textCoordinates.length > 0) {
+        for (let i=0; i<textCoordinates.length; i++) {
+          const { x1, x2, y1, y2, pageNum } = textCoordinates[i];
+          const [page, replacer, region] = await Promise.all([
+            doc.getPage(pageNum),
+            PDFNet.ContentReplacer.create(),
+            PDFNet.Rect.init(x1, y1, x2, y2)
+          ]);
+          replacer.addText(region, lines[i]);
+          replacer.process(page);
+          replacer.destroy();
+        }
+
+        // clear the cache
+        window.docViewer.refreshAll();
+        // update viewer with new document
+        window.docViewer.updateView();
+        // Annotations may contain text so we need to regenerate
+        // our text representation
+        window.docViewer.getDocument().refreshTextData();
+      }
+      // doc.unlock();
+    });
+    closeModal();
+  };
+
+  const changeHandler = e => {
+    setNewText(e.target.value);
+  };
+
+  const keyDownHandler = e => {
+    // (Cmd/Ctrl + Enter)
+    if ((e.metaKey || e.ctrlKey) && e.which === 13) {
+      updateText(e);
+    }
+  };
+
+  const modalClass = classNames({
+    Modal: true,
+    EditTextModal: true,
+    open: isOpen,
+    closed: !isOpen,
+  });
+
+  return isDisabled ? null : (
+    <Swipeable onSwipedUp={closeModal} onSwipedDown={closeModal} preventDefaultTouchmoveEvent>
+      <FocusTrap locked={isOpen}>
+        <div className={modalClass} data-element="editTextModal" onMouseDown={closeModal}>
+          <div className="container" onMouseDown={e => e.stopPropagation()}>
+            <div className="swipe-indicator" />
+            <p>Enter the text you want to replaced</p>
```

Comment:
This should be added to the translation file (and then run the translation script to generate the translation across all languages)

Reply:
The translation has been added